### PR TITLE
Prepare for February 2024 Core Releases

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Prepares the Azure Core libraries for their February 2024 releases.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by mssfang — APPROVED** [review]

File: sdk/core/azure-core-http-netty/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+- Upgraded `azure-core` from `1.45.1` to `1.46.0`.
+- Upgraded `netty-handler` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-handler-proxy` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-buffer` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-codec` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-codec-http` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-codec-http2` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-transport-native-unix-common` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `netty-common` from `4.1.101.Final` to `4.1.106.Final`.
+- Upgraded `reactor-netty-http` from `1.0.39` to `1.0.41`.
```
Comment (srnagar) [inline sdk/core/azure-core-http-netty/CHANGELOG.md:18]: Have we run live tests with the upgraded version of netty-* packages? We should hold off on updating these versions on the day of release.

Comment (alzimmermsft) [reply sdk/core/azure-core-http-netty/CHANGELOG.md]: Rolled this back

File: eng/versioning/external_dependencies.txt [hunk 2]
```diff
@@ -67,8 +67,8 @@ io.netty:netty-tcnative-boringssl-static;2.0.62.Final
 io.netty:netty-transport-native-epoll;4.1.101.Final
 io.netty:netty-transport-native-unix-common;4.1.101.Final
 io.netty:netty-transport-native-kqueue;4.1.101.Final
-io.projectreactor.netty:reactor-netty;1.0.39
-io.projectreactor.netty:reactor-netty-http;1.0.39
+io.projectreactor.netty:reactor-netty;1.0.40
+io.projectreactor.netty:reactor-netty-http;1.0.40
```
Comment (srnagar) [inline eng/versioning/external_dependencies.txt:71]: Do we want to update these versions before release? Are there any live test results that can validate that this new version of reactor-netty* doesn't break anything?

Comment (alzimmermsft) [reply eng/versioning/external_dependencies.txt:71]: We should, we've upgraded Netty and Reactor dependencies which are effectively aligned with Reactor Netty 1.0.40 but it looks like we forgot to upgrade it from 1.0.39

**Review by srnagar — APPROVED** [review]

**Pull request merged**